picard/file.py

```diff
@@ -547,50 +547,37 @@ def _save_images(self, dirname, metadata):
+            new_path = os.path.dirname(new_filename)
+            old_path = os.path.dirname(old_filename)
+            if new_path != old_path:
+                self._apply_additional_files_moves(old_path, new_path, config)
 
     def _compile_move_additional_files_pattern(self, config):
-        patterns = config.setting["move_additional_files_pattern"]
-        pattern_regexes = set()
-        for pattern in patterns.split():
-            pattern = pattern.strip()
-            if not pattern:
-                continue
-            pattern_regex = re.compile(fnmatch.translate(pattern), re.IGNORECASE)
-            match_hidden = pattern.startswith('.')
-            pattern_regexes.add((pattern_regex, match_hidden))
-        return pattern_regexes
-
-    def _get_additional_files_moves(self, old_path, new_path, patterns):
-        moves = set()
-        try:
-            with os.scandir(old_path) as scan:
-                for entry in scan:
-                    is_hidden = entry.name.startswith('.')
-                    for pattern_regex, match_hidden in patterns:
-                        if is_hidden and not match_hidden:
-                            continue
-                        if pattern_regex.match(entry.name):
-                            new_file_path = os.path.join(new_path, entry.name)
-                            moves.add((entry.path, new_file_path))
-                            break  # we are done with this file
-        except OSError as why:
-            log.error("Failed to scan %r: %s", old_path, why)
-        return moves
+        return {
+            (re.compile(fnmatch.translate(pattern), re.IGNORECASE), pattern.startswith('.'))
+            for pattern in set(config.setting["move_additional_files_pattern"].lower().split())
+        }
+
+    def _get_additional_files_moves(self, old_path, new_path, config):
+        patterns = self._compile_move_additional_files_pattern(config)
+        if patterns:
+            try:
+                with os.scandir(old_path) as scan:
+                    for entry in scan:
+                        is_hidden = entry.name.startswith('.')
+                        for pattern_regex, match_hidden in patterns:
+                            if is_hidden and not match_hidden:
+                                continue
+                            if pattern_regex.match(entry.name):
+                                new_file_path = os.path.join(new_path, entry.name)
+                                yield (entry.path, new_file_path)
+                                break  # we are done with this file
+            except OSError as why:
+                log.error("Failed to scan %r: %s", old_path, why)
 
-    def _apply_additional_files_moves(self, moves):
-        for old_file_path, new_file_path in moves:
+    def _apply_additional_files_moves(self, old_path, new_path, config):
+        for old_file_path, new_file_path in self._get_additional_files_moves(old_path, new_path, config):
```

Comment:
Same here, I'd keep the `moves` as a parameter. This method can be nicely tested, especially when you can easily pass a list of move entries. The method also doesn't care about whether `moves` is an iterator or a list, so this eases testing as well.